### DISCUSSION TITLE
Persisting session with client-side routing

### DISCUSSION THREAD

**Discussion by amit08255** [vercel/next.js#12544] [Help]
I am using NextJS with ExpressJS custom server. ExpressJS set up session cookies for my NextJS app. I can get session information in getInitialProps which works well when I am using server-side routing (like normal page load) but when using client-side routing with next/link Link I cannot get session information. Please suggest how I can get session information stored in cookies when using client-side routing and cookies cannot be accessed on client-side.

**Comment by amit08255** [comment]
I need help with this please.

**Comment by rafaelalmeidatk** [comment]
If the cookie is configured with the `httpOnly` flag you are not supposed to have access in the client-side for security reasons. Instead of directly accessing the session cookie you make a request to the server to retrieve the info you need. For example, if you want to retrieve the current logged in user info, you make a request to `https://yourapi/user`, `/me`, etc.
